modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -1104,6 +1272,54 @@ private String _testAddSystemObjectField(
 		return objectDefinitionName;
 	}
 
+	private void _testUpdateCustomObjectField(ObjectField expectedObjectField)
+		throws Exception {
+
+		ObjectField objectField =
+			_objectFieldLocalService.updateCustomObjectField(
+				expectedObjectField.getObjectFieldId(),
+				expectedObjectField.getExternalReferenceCode(),
+				expectedObjectField.getListTypeDefinitionId(),
+				expectedObjectField.getBusinessType(),
+				expectedObjectField.getDBType(),
+				expectedObjectField.getDefaultValue(),
+				expectedObjectField.isIndexed(),
+				expectedObjectField.isIndexedAsKeyword(),
+				expectedObjectField.getIndexedLanguageId(),
+				expectedObjectField.getLabelMap(),
+				expectedObjectField.getName(), expectedObjectField.isRequired(),
+				expectedObjectField.isState(),
+				expectedObjectField.getObjectFieldSettings(), null);
+
+		Assert.assertEquals(
+			expectedObjectField.getExternalReferenceCode(),
+			objectField.getExternalReferenceCode());
```

Comment:
Hi @seltong,

The externalReferenceCode assert should be after of the defaultValue assert.